llvm/utils/TableGen/CodeGenRegisters.cpp

```diff
@@ -1175,24 +1175,21 @@ CodeGenRegBank::CodeGenRegBank(RecordKeeper &Records,
   for (auto &Idx : SubRegIndices)
     Idx.updateComponents(*this);
 
-  // Read in the register definitions.
+  // Read in the register and register tuple definitions.
   std::vector<Record*> Regs = Records.getAllDerivedDefinitions("Register");
-  llvm::sort(Regs, LessRecordRegister());
-  // Assign the enumeration values.
-  for (unsigned i = 0, e = Regs.size(); i != e; ++i)
-    getReg(Regs[i]);
-
-  // Expand tuples and number the new registers.
   std::vector<Record*> Tups =
     Records.getAllDerivedDefinitions("RegisterTuples");
-
   for (Record *R : Tups) {
+    // Expand tuples and merge the vectors
     std::vector<Record *> TupRegs = *Sets.expand(R);
-    llvm::sort(TupRegs, LessRecordRegister());
-    for (Record *RC : TupRegs)
-      getReg(RC);
+    Regs.insert(Regs.end(), TupRegs.begin(), TupRegs.end());
   }
 
+  llvm::sort(Regs, LessRecordRegister());
+  // Assign the enumeration values.
+  for (unsigned i = 0, e = Regs.size(); i != e; ++i)
+    getReg(Regs[i]);
+
```

Comment:
Why switch the order? This doesn't seem related to your change.

Reply:
It's related. `Register` and `RegisterTuple` are all added to same enum in X86GenRegisterInfo.inc. So we need to merge the vectors and sort only once.